### PR TITLE
Bluetooth: Host: Fix ATT PDU alloc being non-blocking.

### PR DESCRIPTION
**Pull request by hermabe**
The API is documented as being blocking. Making it non-blocking was an unintentional API change.

Introduced in https://github.com/zephyrproject-rtos/zephyr/pull/44269.

### CODE REVIEW

File: subsys/bluetooth/host/att.c [hunk 1]
```diff
@@ -542,18 +543,19 @@ struct net_buf *bt_att_chan_create_pdu(struct bt_att_chan *chan, uint8_t op,
 	case ATT_RESPONSE:
 	case ATT_CONFIRMATION:
 		/* Use a timeout only when responding/confirming */
-		buf = bt_l2cap_create_pdu_timeout(NULL, 0, BT_ATT_TIMEOUT);
+		timeout = BT_ATT_TIMEOUT;
```
Comment (hermabe) [inline subsys/bluetooth/host/att.c:546]: One issue here is that the timeout is used for both allocations. Potentially, the total timeout is greater than `BT_ATT_TIMEOUT`.

File: subsys/bluetooth/host/att.c [hunk 2]
```diff
@@ -542,18 +543,19 @@ struct net_buf *bt_att_chan_create_pdu(struct bt_att_chan *chan, uint8_t op,
...
 		break;
 	default:
-		buf = bt_l2cap_create_pdu(NULL, 0);
+		timeout = K_FOREVER;
```
Comment (jori-nordic) [inline subsys/bluetooth/host/att.c:549]: is there a reason to not use `BT_ATT_TIMEOUT` for all allocations ?

Comment (hermabe) [reply subsys/bluetooth/host/att.c:549]: The ATT timeout is just from a request is sent to the response is received. It is ok to wait before sending a request (and the API is documented as blocking).

**Review by jori-nordic — APPROVED** [review]

**Comment by cvinayak** [comment]
I see this (relates to this https://github.com/zephyrproject-rtos/zephyr/issues/25917):
image

And this (seems new!):
image

But does not cause any faults, the modified central_hr continues normal after the error messages.

**Review by jhedberg — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]

**Pull request merged**